packages/dds/tree/src/simple-tree/treeApi.ts

```diff
@@ -142,11 +176,82 @@ export const treeNodeApi: TreeNodeApi = {
  */
-export interface TreeNodeEvents {
+export interface TreeChangeEvents {
 	/**
-	 * Raised on a node right after a change is applied to one of its fields or the fields of a descendant node.
+	 * Emitted by a node when:
+	 *
+	 * - The content of one or more of the node's properties changes (i.e., the property now contains a new node, or
+	 * nothing if it previously contained a node).
+	 *
+	 * - For an array node, when an element is added, removed, or moved.
+	 *
+	 * - For a map node, when a key is added, updated, or removed.
+	 *
+	 * @remarks
+	 * In particular, this event is not raised when:
+	 *
+	 * - Properties of a child node change. Notably, updates to an array node or a map node (like adding or removing
+	 * elements/entries) will raise this event on the array/map node itself, but not on the node that contains the
+	 * array/map node as one of its properties.
+	 *
+	 * - The node is moved to a different location in the tree or removed from the tree.
+	 * In this case the event is raised on the _parent_ node, not the node itself.
+	 *
+	 * Also note that value nodes do not have properties (child nodes), so this event is never raised for them.
+	 *
+	 * For remote edits, this event is not guaranteed to occur in the same order or quantity that it did in
+	 * the client that made the original edit.
+	 * While a batch of edits will as a whole update the tree to the appropriate end state, no guarantees are made about
+	 * how many times this event will be raised during any intermediate states.
+	 * When it is raised, the tree is guaranteed to be in-schema.
+	 *
+	 * @privateRemarks
+	 * This event occurs whenever the apparent contents of the node instance change, regardless of what caused the change.
+	 * For example, it will fire when the local client reassigns a child, when part of a remote edit is applied to the
+	 * node, or when the node has to be updated due to resolution of a merge conflict
+	 * (for example a previously applied local change might be undone, then reapplied differently or not at all).
+	 */
+	afterShallowChange(): void;
+
+	/**
+	 * Emitted by a node when changes happen anywhere in the subtree rooted at it.
+	 *
+	 * @remarks
+	 * The node itself is part of the subtree, so this event will be emitted even if the only changes are to the properties
+	 * of the node itself.
+	 *
+	 * This event is not raised when the node itself is moved to a different location in the tree or removed from the tree.
+	 * In that case it is raised on the _parent_ node, not the node itself.
+	 *
+	 * It may fire at a time when the change(s) that triggered it are not yet visible if the listener inspects the tree.
+	 * In that case, it is guaranteed to fire again after the change(s) _are_ visible to the listener.
+	 *
+	 * Also note that value nodes do not have properties (child nodes), so this event is never raised for them.
+	 *
+	 * For remote edits, this event is not guaranteed to occur in the same order or quantity that it did in
```

Comment:
```suggestion
```

You can't register this event on them via the simple-tree API anyway.

Reply:
This came from Craig's original proposal, I think it's meant to convey that the firings of this event that happened on the client that originally made the edit will not necessarily match the ones you get when you receive the edit. Seems useful to clarify that, but maybe there's a better phrasing for it?

Reply:
I meant to remove line 248, sorry.